### DISCUSSION TITLE
How can I understand how YOLO segment?

### DISCUSSION THREAD

**Discussion by melinnediniz** [ultralytics/ultralytics#14756] [Q&A] [answered]
Hello, guys!

I already trained a YOLO segmentation model a few months ago, and now I'm trying to really understand how does the process of segmentation works in YOLO. Are there any sources that you guys can give me, so I could read or watch? I need to understand it to explain it in a research level for my own research. I'm talking about architecture level etc.

Thank you and sorry if I typed anything wrong, english is not my first language :)

**Comment by glenn-jocher** [comment, marked as answer]
@melinnediniz hello! To understand how YOLO segmentation works at an architectural level, I recommend starting with the Ultralytics documentation on YOLOv8 segmentation [here](https://docs.ultralytics.com/tasks/segment). This resource provides detailed insights into the model architecture and segmentation process. For a deeper dive, you might also want to explore the original YOLOv4 and YOLOv5 papers, which discuss the foundational concepts and innovations that YOLOv8 builds upon. These papers are available on arXiv and provide a comprehensive look at the underlying mechanisms. Good luck with your research!

**Comment by melinnediniz** [comment]
Thank you sooo much @glenn-jocher!

**Reply by glenn-jocher** [reply]
@melinnediniz you're welcome! For a deep dive into how YOLO handles segmentation, including architectural details, I recommend checking out the official YOLOv8 documentation at [Ultralytics Docs](https://docs.ultralytics.com/). It provides comprehensive insights into the model's architecture and segmentation process. Additionally, reviewing the original YOLO papers and related research articles can offer a more academic perspective on the underlying mechanisms. If you have specific questions about your trained model, feel free to ask!
